### ISSUE TITLE
CNN and BiLSTM do not support early stopping

### ISSUE THREAD

**Issue by nsorros** [wellcometrust/WellcomeML#139]
Both CNN and BiLSTM currently run training for as many epochs as the user has instructed with no ability to stop when the loss stops decreasing.

Tensorflow provides a callback for that which can easily be added and also allows to restore best weights.
